modules/cf-core/src/protocol/utils/get-outcome-increments.ts

```diff
@@ -84,14 +86,19 @@ export async function computeTokenIndexedFreeBalanceIncrements(
 async function handleRefundAppOutcomeSpecialCase(
   encodedOutcome: string,
   appInstance: AppInstance,
-  provider: BaseProvider
+  provider: BaseProvider,
+  blockNumberToUseIfNecessary?: number
 ): Promise<TokenIndexedCoinTransferMap> {
   let mutableOutcome = encodedOutcome;
   let attempts = 1;
   while (attempts <= 10) {
     const [{ to, amount }] = decodeRefundAppState(mutableOutcome);
-
-    if (amount.gt(0)) {
+    const currentBlockNumber = await provider.getBlockNumber();
+    if (
+      amount.gt(0) ||
+      (blockNumberToUseIfNecessary &&
+        currentBlockNumber >= blockNumberToUseIfNecessary)
+    ) {
       return {
         [(appInstance.state as CoinBalanceRefundState).tokenAddress]: {
           [to]: amount
```

Comment:
I think this is right, but it would benefit from declaring some variables with names, or adding some comments, as it reads confusingly.